test/extensions/filters/http/lua/lua_filter_test.cc

```diff
@@ -181,7 +181,7 @@ class LuaHttpFilterTest : public testing::Test {
     function envoy_on_request(request_handle)
       request_handle:logTrace(request_handle:headers():get(":path"))
 
-      if request_handle:body() ~= nil then
+      if request_handle:body():length() ~= 0 then
```

Comment:
I'm a little worried about this change breaking scripts, but I'm not sure how likely that is. I guess with a runtime guard we can see how it goes? The alternative would be to allow `body()` to take an optional boolean which enables the behavior? This could default to the existing behavior? I think I'm probably leaning towards that now that I think about it? @dio WDYT?

Reply:
Yes, this PR introduces a new runtime feature flag (and tested in integration test).

But yeah, thinking it again, to have a cleaner approach: seems like bool (or enum (`Envoy.ALWAYS_WRAP_BODY`)? for syntactic sugar, similar to [this PR](https://github.com/envoyproxy/envoy/pull/14522/files#diff-d9748f1e953ede2dcdd1ebbb9141ebb079f2e208ab701dff9b1d08ba7f4486edR2293)) or a new API (e.g. `wrappedBody()`, this new API always has a non-nil "buffer").

```lua

-- with bool.
response_handle:body(true):setBytes("ok")

-- with "enum".
response_handle:body(Envoy.ALWAYS_WRAP_BODY):setBytes("ok")

-- with new API.
response_handle:wrappedBody():setBytes("ok")
```

Reply:
I will vote for the first one, because:
1. argument is enough to distinguish them, not need to add an extra method
2. there is only "always wrap body" and "not always wrap body", so a boolean is enough.

Reply:
SG. The reason for offering new API is for readability in the Lua code.